### DISCUSSION TITLE
KONG 3.0.0 - http.headers.X-Consumer-Username missing in header from request routed

### DISCUSSION THREAD

**Discussion by kamalnagpalsirion** [Kong/kong#12722] [Help] [answered]
Hi,

I am using KONG 3.0.0 and facing the issue related to upstream headers http.headers.X-Consumer-Username set by KONG.
I have created proxy reference of path in the route and not getting the header.If I am removing the proxy reference then its working fine and I am getting the header specified.

Service:
`{
  "write_timeout": 60000,
  "connect_timeout": 60000,
  "protocol": "http",
  "tls_verify": null,
  "tls_verify_depth": null,
  "host": "myserver",
  "tags": null,
  "updated_at": 1699960024,
  "retries": 5,
  "id": "e2d9336f-bbb7-43d1-bd22-d9d1cd4w2547",
  "enabled": true,
  "name": "test-api",
  "port": 8091,
  "ca_certificates": null,
  "client_certificate": null,
  "read_timeout": 60000,
  "created_at": 1641450512,
  "path": null,
  "extras": {}
}`

Route Create as below:

`{
  "request_buffering": true,
  "response_buffering": true,
  "service": {
    "id": "e2d9336f-bbb7-43d1-bd22-d9d1cd4w2547"
  },
  "https_redirect_status_code": 426,
  "preserve_host": false,
  "path_handling": "v1",
  "created_at": 1709916769,
  "updated_at": 1710149639,
  "hosts": [
    "client.mycloud.io"
  ],
  "strip_path": false,
  "id": "e3bc49e7-6b4a-4884-9b50-fa280bbf02x2",
  "name": null,
  "tags": null,
  "sources": null,
  "paths": [
    "/api/v2/customer/CL\d+/payroll",
    "/api/v2/customer/CL\d+/data"
  ],
  "methods": [
    "GET"
  ],
  "headers": null,
  "destinations": null,
  "snis": null,
  "regex_priority": 0,
  "protocols": [
    "http"
  ]
}`

I have enabled the ACL and oauth plugin on the specified route.
But when I am triggering the request from Postman, on my server I am not getting the header of Username which I have used for checking the details of user passing the request.

Please help.

**Comment by hanshuebner** [comment]
Hello @kamalnagpalsirion,

> I have created proxy reference of path in the route and not getting the header.If I am removing the proxy reference then its working fine and I am getting the header specified.

How exactly do you create the proxy reference in the route?

Also note that Kong 3.0.0 is no longer supported.  Please consider upgrading.

-Hans

**Reply by kamalnagpalsirion** [reply, marked as answer]
Hi Hans,

I created the proxy reference in the route path as "/api/v2/customer/CL\d+/payroll".After referring the documentation [https://docs.konghq.com/gateway/latest/how-kong-works/routing-traffic/#request-path], I realised that I have put the ~ in the path to treat it as regex.I don't know how I missed it but I should have created the path as "~/api/v2/customer/CL\d+/payroll" which resolved the issue.

Thanks for the response.

-Kamal
